### DISCUSSION TITLE
Reg: Scaling of Http connections for large number of concurrent connections

### DISCUSSION THREAD

**Discussion by bazooka720** [HoneyryderChuck/httpx#57] [General]
I had a question and was hoping to get thoughts about usage of the gem. We are needing to enable our tenants/users to connect to a data platform/engine through http. Issue is once connection is estabished ...many queries need to be ideally run through this connection. The engine is actually a k8s custer and each tenant can have 1 or more such engines and so we define a conneciton for each "user" and the "cluster". Each user can have more than 1 potential customer and there can be hundreds or even thousands of such users and thus the connections. We were hoping to use an approach that will allow easy horizontal scaling and ideally take the load of the app itself where possible. Curious ...if for use case like this your gem might work? Has gem been used in prod with complex and heavy workloads?

**Comment by HoneyryderChuck** [comment]
Hi, based on your description,  httpx should cover your needs. If you need to reuse connections, you should use the persistent plugin. Be sure to read the wiki (link in the readme) for use cases, examples and descriptions.

**Comment by HoneyryderChuck** [comment]
Don't know much about presto or trino. It seems it's using faraday, and httpx ships with a faraday adapter. You should look in Torino socs on how to switch the adapter, and if not, tell them you'd be interested in having the option.
